### PR TITLE
Reland: Deprecate passing EM_CONFIG without a file

### PR DESCRIPTION
**Pull request by sbc100**
This is feature that I don't think has any users and removing
can simplify out code and testing matrix.

It was originally added back 2012 when it seems that per-project
configuration was a desirable feature:
99c46475c1944c1e62e04f30bd0db3233639515d

I think these days this should not be necessary, and even if it
if we still support `--em-config` on the command line.

Original PR: #11202

### CODE REVIEW

File: tools/shared.py [hunk 1]
```diff
@@ -64,6 +64,7 @@
 diagnostics.add_warning('linkflags')
 diagnostics.add_warning('emcc')
 diagnostics.add_warning('undefined', error=True)
+diagnostics.add_warning('deprected')
```
Comment (kripken) [inline tools/shared.py:67]:
```suggestion
diagnostics.add_warning('deprecated')
```

File: tests/test_sanity.py [hunk 2]
```diff
@@ -433,7 +433,6 @@ def test_em_config_env_var(self):
       }
     ''')
 
-    wipe()
```
[the remark this answers is not in the document]
Comment (sbc100) [reply tests/test_sanity.py:436]: Done

**Comment by sbc100** [comment]
Should be good to go now

File: ChangeLog.md [hunk 3]
```diff
@@ -18,6 +18,11 @@ See docs/process.md for how version tagging works.
 Current Trunk
 -------------
 
+- The `EM_CONFIG` environment variable and `--em-config` command line option no
+  longer support a literal python string. Instead the name of a config file is
+  required. Since all config file settings are individually override-able using
+  `EM_FOO` this should be needed.
```
Comment (kripken) [inline ChangeLog.md:24]:
```suggestion
  `EM_FOO` this should be enough.
```

File: tools/gen_struct_info.py [hunk 4]
```diff
@@ -396,7 +396,7 @@ def inspect_code(headers, cpp_opts, structs, defines):
   # -Oz optimizes enough to avoid warnings on code size/num locals
   cmd = [shared.EMCC] + cpp_opts + ['-o', js_file[1], src_file[1],
                                     '-O0', '--js-opts', '0', '--memory-init-file', '0',
-                                    '-Werror', '-Wno-format',
+                                    '-Werror', '-Wno-format', '-Wno-deprecated',
```
Comment (kripken) [inline tools/gen_struct_info.py:399]: Oh, this uses the literal form - so we'll need to change that eventually?

Comment (sbc100) [reply tools/gen_struct_info.py:399]: No, i does not use it directly, but we have tests the verify the EM_CONFIG in the environment works, and that everything that emcc runs internally too.

Comment (kripken) [reply tools/gen_struct_info.py:399]: So this change is just for testing purposes? Maybe a comment then, so it is obvious why it's there and when we can remove it?

**Pull request merged**